backend/src/routes/api/notebooks/notebookUtils.ts

```diff
@@ -189,6 +189,26 @@ export const patchNotebook = async (
     .then((response) => response.body as Notebook);
 };
 
+export const replaceNotebook = async (
+  fastify: KubeFastifyInstance,
+  body: Notebook,
+  namespace: string,
+  notebookName: string,
+): Promise<Notebook> => {
+  return fastify.kube.customObjectsApi
+    .replaceNamespacedCustomObject(
+      'kubeflow.org',
+      'v1',
+      namespace,
+      'notebooks',
+      notebookName,
+      body,
+      undefined,
+      undefined,
+      undefined,
+    )
+    .then((response) => response.body as Notebook);
```

Comment:
What happens if the replace fails? (i would at least log the error, not sure if we would send a descriptive error)

Reply:
Currently a 422 error gets returned (if the object that the frontend sent is incorrect, otherwise the usual codes). We could theoretically take the message of the openshift error to let the person know which part of the patch failed but that doesn't seem to be done for the rest of the functions.